### ISSUE TITLE
Preventing ancient dates from match by compromise-dates plugin

### ISSUE THREAD

**Issue by thegoatherder** [spencermountain/compromise#868]
Found a _hmmm_ bug in `compromise-dates` where it's matching an invalid date in format `dd/mm/{invalid 3-digit year}`

**Steps to reproduce:**

```
nlp(`30/01/194`).dates().out('array')
```

**Expected:**

`[]`

**Actual:**

`["30/01/194"]`

**Discussion**

I thought this was a bug at first, but then I realised it's valid for the year 194 AD. In my case, the date is malformed caused by a typo. I know this as it's a Date of Birth of someone who is still alive!  The reason I thought this was a bug was that it caused an error in SQL Server:

```sql
DECLARE @date date
SET @date = '1942-02-01'  -- works
SET @date = '194-02-01'    -- throws: Conversion failed when converting date and/or time from character string.
```

The minimum date storable by SQL Server is `January 1st, 1753` ([see source](https://docs.microsoft.com/en-us/dotnet/api/system.data.sqltypes.sqldatetime.minvalue?view=net-5.0#:~:text=SqlDateTime-,Remarks,structure%20is%20January%201%2C%201753.))

Can we initialise `compromise-dates` with a `minDate` and `maxDate` to prevent it picking up dates that are out of range? Or should we just tackle this in post-processing?

Thanks!

**Comment by spencermountain** [comment]
hah, yeah good point. I agree that example shouldn't parse in the compromise-date plugin. Maybe we should tighten-up the regex.
you should be able to do a .filter() for now, but will keep this open
cheers

**Comment by thegoatherder** [comment]
Thanks. On my end I've gone ahead and added an `isPermitted()` method for the time being:

```js
const spacetime = require('spacetime')
const globalMin = spacetime.now('Europe/London').minus(150, 'years')
const globalMax = spacetime.now('Europe/London').add(50, 'years')

spacetime.extend({
  isPermitted: function() {
    return this.isBetween(globalMin, globalMax, true)
  }
})
module.exports = spacetime
```
This does the trick... but really I would prefer for `isValid()` to return false if the date is outside allowable boundaries, to save having to do `myDate.isValid() && myDate.isPermitted()`

**Comment by spencermountain** [comment]
fixed in 14.5.0
